pdns/misc.cc

```diff
@@ -1485,3 +1485,36 @@ std::vector<ComboAddress> getResolvers(const std::string& resolvConfPath)
 
   return results;
 }
+
+size_t getPipeBufferSize(int fd)
+{
+#ifdef F_GETPIPE_SZ
+  int res = fcntl(fd, F_GETPIPE_SZ);
+  if (res == -1) {
+    return 0;
+  }
+  return res;
+#else
+  errno = ENOSYS;
+  return 0;
+#endif /* F_GETPIPE_SZ */
+}
+
+bool setPipeBufferSize(int fd, size_t size)
```

Comment:
Why a `size_t` ? 
`argsvMap.asNum()` returns an `int`, and the `fcntl` takes an `int` as well.

Reply:
I used a `size_t` because a negative value makes no sense here. `asNum()` and `fcntl()` are generic interfaces, but IMHO that does not mean we can't use a more correct type.

Reply:
I'm happy with the unsigned part of size_t, but negative numbers will be converted to very large values by the int to size_t conversion. We probably need an asSize() or similar that rejects negative values.

Reply:
That would be awesome, yes!